include/aws/common/math.gcc_overflow.inl

```diff
@@ -120,6 +120,20 @@ AWS_STATIC_IMPL uint32_t aws_add_u32_saturating(uint32_t a, uint32_t b) {
     return res;
 }
 
+/**
+ * Search from the MSB to LSB, looking for a 1
+ */
+AWS_STATIC_IMPL size_t aws_count_leading_zeroes(int32_t n) {
+    return __builtin_clz(n);
```

Comment:
1) seems weird to take a signed argument.
2) should we implement u32/u64/size variants while we're in here?
if you don't want to add/test the other two variants, at least name this `_u32` for now
3) [trivial] largest possible return value is 64, worth returning uint8_t or just weird?


actual builtin signatures are:
```
int __builtin_clz (unsigned int x)
int __builtin_clzl (unsigned long)
int __builtin_clzll (unsigned long long)

unsigned char _BitScanForward(
   unsigned long * Index,
   unsigned long Mask
);
unsigned char _BitScanForward64(
   unsigned long * Index,
   unsigned __int64 Mask
);
```

Reply:
Implemented i32, u32, i64, u64, size